tests/unit/api/main.py

```diff
@@ -1247,16 +1247,34 @@ def test_switch_nexist(self, db):
 
     def test_port_nexist(self, db):
         """Removing a port that does not exist should report the error"""
-        api.switch_register('sw0', type=MOCK_SWITCH_TYPE,
+        api.switch_register('sw0', type=MOCK_SWITCH_TYPE, 
 		username="switch_user", password="switch_pass", hostname="switchname")
         with pytest.raises(api.NotFoundError):
             api.switch_delete_port('sw0', '5')
 
+class Test_list_switches:
+
+    def test_list_switches(self, db):
+        assert json.loads(api.list_switches()) == []
+        api.switch_register('sw0', type=MOCK_SWITCH_TYPE, 
+		username="foo", password="bar", hostname="baz")
+        assert db.query(model.Switch).one().label == 'sw0'
+        assert json.loads(api.list_switches()) == ['sw0']
+
+        api.switch_register('mock', type=MOCK_SWITCH_TYPE, 
+		username="user", password="password", hostname="host")
+        api.switch_register('cirius', type=MOCK_SWITCH_TYPE, 
+		username="user", password="password", hostname="switch")
+        assert sorted(json.loads(api.list_switches())) == [
```

Comment:
@ron-unrau , I think if you sort it in the code, as Sahil suggested earlier, there is no need to sort it here.  I don't have any strong opinion on the sorting stuff, but I think its better to put in the code rather than a test case.